test/cmd/diff.sh

```diff
@@ -44,6 +47,11 @@ run_kubectl_diff_tests() {
     # Test that we have a return code bigger than 1 if there is an error when diffing
     kubectl diff -f hack/testdata/invalid-pod.yaml || test $? -gt 1
 
+    # Test diff with server-side apply
+    kubectl apply -f hack/testdata/pod.yaml --server-side --force-conflicts
+    # Make sure that diffing the resource right after returns nothing (0 exit code).
+    kubectl diff -f hack/testdata/pod.yaml --server-side
```

Comment:
Oh this produces a diff and exits w/ 1 only because of the field manager timestamp

```diff
diff -u -N /tmp/LIVE-571109069/v1.Pod.namespace-1583178564-24022.test-pod /tmp/MERGED-959055816/v1.Pod.namespace-1583178564-24022.test-pod
--- /tmp/LIVE-571109069/v1.Pod.namespace-1583178564-24022.test-pod	2020-03-02 19:49:27.028935894 +0000
+++ /tmp/MERGED-959055816/v1.Pod.namespace-1583178564-24022.test-pod	2020-03-02 19:49:27.032936239 +0000
@@ -22,7 +22,7 @@
             f:name: {}
     manager: kubectl
     operation: Apply
-    time: "2020-03-02T19:49:26Z"
+    time: "2020-03-02T19:49:27Z"
   - apiVersion: v1
     fieldsType: FieldsV1
     fieldsV1:
```

Reply:
@jennybuckley This happens only because we've changed the object on a no-op apply. That's a bad bug. Don't we have a test for that?

Reply:
cc @jennybuckley

Reply:
The test is only to verify no-op updates don't write to etcd.